api/v1alpha1/phschedule_types.go

```diff
@@ -19,40 +19,44 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-
 // +kubebuilder:validation:Enum=inactive;daily;weekly;monthly;custom
 type RecurrenceType string
 
 const (
 	RecurrenceTypeInactive RecurrenceType = "inactive"
-	RecurrenceTypeDaily RecurrenceType = "daily"		// 0 4 * * *
-	RecurrenceTypeWeekly RecurrenceType = "weekly"		// 0 4 * * 0
-	RecurrenceTypeMonthly RecurrenceType = "monthly"	// 0 4 1 * *
-	RecurrenceTypeCustom RecurrenceType = "custom"
+	RecurrenceTypeDaily    RecurrenceType = "daily"   // 0 4 * * *
+	RecurrenceTypeWeekly   RecurrenceType = "weekly"  // 0 4 * * 0
+	RecurrenceTypeMonthly  RecurrenceType = "monthly" // 0 4 1 * *
+	RecurrenceTypeCustom   RecurrenceType = "custom"
 )
 
 type Recurrence struct {
 	Type RecurrenceType `json:"type"`
-	Cron string `json:"cron,omitempty"`
+	Cron string         `json:"cron,omitempty"`
 }
 
 // PhScheduleSpec defines the desired state of PhSchedule
 type PhScheduleSpec struct {
-	Recurrence Recurrence `json:"recurrence"`
+	Recurrence  Recurrence        `json:"recurrence"`
 	JobTemplate PhJobTemplateSpec `json:"jobTemplate"`
 }
 
 // PhScheduleStatus defines the observed state of PhSchedule
 type PhScheduleStatus struct {
-	Valid		bool   `json:"valid,omitempty"`
+	Valid       bool         `json:"valid,omitempty"`
 	NextRunTime *metav1.Time `json:"nextRunTime,omitempty"`
+	Message     string       `json:"message,omitempty"`
+	Active      bool         `json:"active,omitempty"`
 }
```

Comment:
Can active be removed? Because we can know if it is active by the `spec.recurrence.type`

And can we change the field `Valid` to `Invalid`. Because the default value of boolean type is false. I prefer to make the default value is `Invalid=false` rather than `Valid=false1`. It makes the new UI to render the list after creating a schedule easier to implement.

Reply:
BTW, I found in the current design, active and inactive is not a switch. Therefore, it cannot remember the Cron settings when it switches from active to inactive and then active.

Reply:
@jackklpan since we remove "active"
do we still need a switch to record the Cron setting?
Also, I am not sure I get your idea here, can you elaborate the function in more detail?

Reply:
@popcornylu  active removed

Reply:
Sorry, I mean in UI we are not using a switch or checkbox.
So, schedules cannot remember the "custom recurrence" pattern.
Gitlab uses a checkbox in UI to represent "active". They can remember Cron settings.
Therefore, it's a UI issue. Not related to the current code implementation~

Reply:
ok, got it